pkg/scheduler/framework/framework_test.go

```diff
@@ -702,3 +702,307 @@ func TestRunFilterPlugins(t *testing.T) {
+				&DummyAllPurposePlugin{
+					name: dummyPreFilterPluginNameA,
+					preFilterRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (status *Status) {
+						return nil
+					},
+				},
+				&DummyAllPurposePlugin{
+					name: dummyPreFilterPluginNameB,
+					preFilterRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot) (status *Status) {
+						return nil
+					},
+				},
+			},
+			filterPlugins: []FilterPlugin{
+				&DummyAllPurposePlugin{
+					name: dummyFilterPluginNameA,
+					filterRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot, cluster *fleetv1beta1.MemberCluster) (status *Status) {
+						if cluster.Name == altClusterName {
+							return NewNonErrorStatus(ClusterUnschedulable, dummyFilterPluginNameA)
+						}
+						return nil
+					},
+				},
+				&DummyAllPurposePlugin{
+					name: dummyFilterPluginNameB,
+					filterRunner: func(ctx context.Context, state CycleStatePluginReadWriter, policy *fleetv1beta1.ClusterSchedulingPolicySnapshot, cluster *fleetv1beta1.MemberCluster) (status *Status) {
+						if cluster.Name == anotherClusterName {
+							return NewNonErrorStatus(ClusterUnschedulable, dummyFilterPluginNameB)
+						}
+						return nil
+					},
+				},
+			},
+			wantScored: ScoredClusters{
+				{
+					Cluster: &clusters[0],
+					Score:   &ClusterScore{},
+				},
+			},
+			wantFiltered: []*filteredClusterWithStatus{
+				{
+					cluster: &clusters[1],
+					status:  NewNonErrorStatus(ClusterUnschedulable, dummyFilterPluginNameA),
+				},
+				{
+					cluster: &clusters[2],
+					status:  NewNonErrorStatus(ClusterUnschedulable, dummyFilterPluginNameB),
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			profile := NewProfile(dummyProfileName)
+			for _, p := range tc.preFilterPlugins {
+				profile.WithPreFilterPlugin(p)
+			}
+			for _, p := range tc.filterPlugins {
+				profile.WithFilterPlugin(p)
+			}
+			f := &framework{
+				profile:      profile,
+				parallelizer: parallelizer.NewParallelizer(parallelizer.DefaultNumOfWorkers),
+			}
+
+			ctx := context.Background()
+			state := NewCycleState()
+			scored, filtered, err := f.runAllPluginsForPickAllPlacementType(ctx, state, policy, clusters)
+			if tc.expectedToFail {
+				if err == nil {
+					t.Errorf("runAllPluginsForPickAllPlacementType(), want error")
+				}
+				return
+			}
+
+			// The method runs in parallel; as a result the order cannot be guaranteed.
+			// Sort the results by cluster name for comparison.
+			lessFuncScoredCluster := func(scored1, scored2 ScoredCluster) bool {
+				return scored1.Cluster.Name < scored2.Cluster.Name
+			}
+			lessFuncFilteredCluster := func(filtered1, filtered2 *filteredClusterWithStatus) bool {
+				return filtered1.cluster.Name < filtered2.cluster.Name
+			}
```

Comment:
nit, create a variable for these two func?

Reply:
Done.